plugins/scaffolder-backend/src/lib/templating/SecureTemplater.test.ts

```diff
@@ -0,0 +1,147 @@
+    const renderWith = await SecureTemplater.loadRenderer({
+      cookiecutterCompat: true,
+    });
+    const renderWithout = await SecureTemplater.loadRenderer();
+
+    // Same two tests repeated to make sure switching back and forth works
+    expect(renderWith('{{ 1 | jsonify }}', {})).toBe('1');
+    expect(renderWith('{{ 1 | jsonify }}', {})).toBe('1');
+    expect(() => renderWithout('${{ 1 | jsonify }}', {})).toThrow(
+      /Error: filter not found: jsonify/,
+    );
+    expect(renderWith('{{ 1 | jsonify }}', {})).toBe('1');
+    expect(() => renderWithout('${{ 1 | jsonify }}', {})).toThrow(
+      /Error: filter not found: jsonify/,
+    );
+    expect(() => renderWithout('${{ 1 | jsonify }}', {})).toThrow(
+      /Error: filter not found: jsonify/,
+    );
+    expect(() => renderWithout('${{ 1 | jsonify }}', {})).toThrow(
+      /Error: filter not found: jsonify/,
+    );
+    expect(renderWith('{{ 1 | jsonify }}', {})).toBe('1');
+  });
+
+  it('should make parseRepoUrl available when requested', async () => {
+    const parseRepoUrl = jest.fn(() => ({
+      repo: 'my-repo',
+      owner: 'my-owner',
+      host: 'my-host.com',
+    }));
+    const renderWith = await SecureTemplater.loadRenderer({ parseRepoUrl });
+    const renderWithout = await SecureTemplater.loadRenderer();
+
+    const ctx = {
+      repoUrl: 'https://my-host.com/my-owner/my-repo',
+    };
+
+    expect(renderWith('${{ repoUrl | parseRepoUrl | dump }}', ctx)).toBe(
+      JSON.stringify({
+        repo: 'my-repo',
+        owner: 'my-owner',
+        host: 'my-host.com',
+      }),
+    );
+    expect(renderWith('${{ repoUrl | projectSlug }}', ctx)).toBe(
+      'my-owner/my-repo',
+    );
+    expect(() =>
+      renderWithout('${{ repoUrl | parseRepoUrl | dump }}', ctx),
+    ).toThrow(/Error: filter not found: parseRepoUrl/);
+    expect(() => renderWithout('${{ repoUrl | projectSlug }}', ctx)).toThrow(
+      /Error: filter not found: projectSlug/,
+    );
+
+    expect(parseRepoUrl.mock.calls).toEqual([
+      ['https://my-host.com/my-owner/my-repo'],
+      ['https://my-host.com/my-owner/my-repo'],
+    ]);
+  });
+
+  it('should not allow helpers to be rewritten', async () => {
+    const render = await SecureTemplater.loadRenderer({
+      parseRepoUrl: () => ({
+        repo: 'my-repo',
+        owner: 'my-owner',
+        host: 'my-host.com',
+      }),
+    });
+
+    const ctx = {
+      repoUrl: 'https://my-host.com/my-owner/my-repo',
+    };
+    expect(
+      render(
+        '${{ ({}).constructor.constructor("parseRepoUrl = () => JSON.stringify(`inject`)")() }}',
+        ctx,
+      ),
+    ).toBe('');
+
+    expect(render('${{ repoUrl | parseRepoUrl | dump }}', ctx)).toBe(
+      JSON.stringify({
+        repo: 'my-repo',
+        owner: 'my-owner',
+        host: 'my-host.com',
+      }),
+    );
+  });
+
+  it('allows pollution during a single template execution', async () => {
```

Comment:
looks funny to have a test for it, but of course it's both useful and informative :)